planner/core/logical_plan_builder.go

```diff
@@ -1994,6 +1995,17 @@ func (b *PlanBuilder) buildDataSource(tn *ast.TableName) (LogicalPlan, error) {
 
 	if tableInfo.GetPartitionInfo() != nil {
 		b.optFlag = b.optFlag | flagPartitionProcessor
+		// check partition
+		if len(tn.PartitionNames) != 0 {
```

Comment:
this check can be removed, the following code block can be simplified to:
```go
for _, name := range tn.PartitionNames {
	// find partition by name
}
```

Reply:
done.